### PR TITLE
Feature/package

### PR DESCRIPTION
**Pull request by ggsdc**
Tested it with external app and alone and it should work. Just have to set up the deployment of cornflow-server as a package. Probabily is going to be called just cornflow as it installs core and client as well.

After we could ditch the core part and merge it back to cornflow as it is a library now for any external app as well (#370)

### CODE REVIEW

File: cornflow-server/cornflow/app.py [hunk 1]
```diff
@@ -72,9 +72,9 @@ def _fk_pragma_on_connect(dbapi_con, con_record):
     for res in resources:
         api.add_resource(res["resource"], res["urls"], endpoint=res["endpoint"])
 
-    docs = FlaskApiSpec(app)
-    for res in resources:
-        docs.register(target=res["resource"], endpoint=res["endpoint"])
+    # docs = FlaskApiSpec(app)
+    # for res in resources:
+    #     docs.register(target=res["resource"], endpoint=res["endpoint"])
```
Comment (ggsdc) [inline cornflow-server/cornflow/app.py:77]: This is the only that should be reviewed. I have not been able to run it with flask api swagger active.

File: cornflow-server/cornflow/endpoints/__init__.py [hunk 2]
```diff
@@ -205,6 +205,6 @@
         endpoint="licences",
     )
 ]
-
-if len(external_resources):
-    resources = resources + external_resources
+#
+# if len(external_resources):
+#     resources = resources + external_resources
```
Comment (ggsdc) [inline cornflow-server/cornflow/endpoints/__init__.py:212]: All references here to external_app can be deleted now

File: cornflow-server/cornflow/init_service.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
+import os
```
Comment (ggsdc) [inline cornflow-server/cornflow/init_service.py]: @ampuerin This is the new entrypoint

File: cornflow-server/Dockerfile [hunk 4]
```diff
@@ -26,19 +26,12 @@ ENV PYTHONUNBUFFERED 1
...
-COPY ./requirements.txt /usr/src/app/requirements.txt
-RUN pip install -r requirements.txt --force-reinstall
-
-# copy project
-COPY cornflow /usr/src/app/cornflow
-COPY docs /usr/src/app/docs
-COPY migrations /usr/src/app/migrations
-COPY examples /usr/src/app/examples
-COPY *.py /usr/src/app/
+RUN pip install "git+https://github.com/baobabsoluciones/cornflow@feature/package#subdirectory=cornflow-server"
```
Comment (ggsdc) [inline cornflow-server/Dockerfile:29]: This has to be changed before merging and we have to add here a cornflow version code

**Review by ampuerin — COMMENTED** [review]
For do operations with .ssh folder we need to create it before. In Dockerfile it will be need:

```
# create folder for custom ssh keys
RUN mkdir /usr/src/app/.ssh
```

File: cornflow-server/cornflow/cli/init_cornflow_service.py [hunk 5]
```diff
@@ -148,12 +148,22 @@ def init_cornflow_service():
             f"Starting cornflow + {os.getenv('EXTERNAL_APP_MODULE', '../external_app')}"
         )
         os.chdir("/usr/src/external_app")
+
+        if register_key():
+            github_host = os.getenv("github_host", None)
```
Comment (ampuerin) [inline cornflow-server/cornflow/cli/init_cornflow_service.py:153]: `"github_host"` -> better in capital letters, right?

`github_host = os.getenv("GITHUB_HOST", None)`

Comment (ggsdc) [reply cornflow-server/cornflow/cli/init_cornflow_service.py:153]: Pycharm prefers if they are small caps

**Review by marioncottard — DISMISSED** [review]

**Comment by ggsdc** [comment]
I will close for now so the test do not get run all the time until I have finished all the changes needed

**Comment by ggsdc** [comment]
PRE-MERGE TASKLIST:

- [ ] Update cornflow version on Dockerfile.
- [ ] Update cornflow version on setup.py
- [ ] Modify workflow to add a tag after deployment on pypi so dockerhub can launch.
- [ ] Finish unit testing of cli

File: cornflow-server/setup.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+required = []
+with open("requirements.txt", "r") as fh:
+    required.append(fh.read().splitlines())
+
+setuptools.setup(
+    name="cornflow",
+    version="1.0.0a6",
+    author="baobab soluciones",
+    author_email="sistemas@baobabsoluciones.es",
```
Comment (ampuerin) [inline cornflow-server/setup.py:14]: I think it's better to contact with cornflow group at cornflow@baobabsoluciones.es

File: cornflow-server/Dockerfile [hunk 7]
```diff
@@ -8,6 +8,9 @@ LABEL maintainer="sistemas@baobabsoluciones"
 ENV DEBIAN_FRONTEND noninteractive
 ENV TERM linux
 
+# CORNFLOW vars
+ARG CORNFLOW_VERSION=1.0.0a6
+
 # install linux pkg
 RUN apt update -y && apt-get install -y --no-install-recommends \
```
Comment (ampuerin) [inline cornflow-server/Dockerfile:15]: here it's necessary to install ssh

`RUN apt update -y && apt-get install -y --no-install-recommends gcc ssh`

**Review by marioncottard — APPROVED** [review]

**Pull request merged**